### ISSUE TITLE
connectionPool setBindRequest ignored in some cases

### ISSUE THREAD

**Issue by diegogranados** [pingidentity/ldapsdk#43]
Hi, 
I was trying to use credentials reconfiguration in a connected connection pool using setBindRequest support introduced in ldapsdk 4.0.6, but found a small issue. 

The new bind credentials are only used when the connection pool is created with some constructors. Specifically, a pool created using the constructor

public LDAPConnectionPool(final LDAPConnection connection,
                            final int initialConnections,
                            final int maxConnections,
                            final int initialConnectThreads,
                            final PostConnectProcessor postConnectProcessor,
                            final boolean throwOnConnectFailure)

does not honor the new credentials set via setBindRequest (i.e. keeps using the original credentials from the passed connection for new connections). I looked at the code and saw that the internally created ServerSet keeps a reference to the original bindRequest that is not updated on setBindRequest calls, but didn't verify this is indeed the problem)

The problem is not triggered when the pool was created using the constructor which receives ServerSet and BindRequest arguments (I am currently this as a workaround)

**Comment by dirmgr** [comment]
Thank you very much for reporting this, and I apologize for having taken all day to get to it. I did confirm that it's a problem, and I have committed a fix for it (https://github.com/pingidentity/ldapsdk/commit/7904e5585055c16ec8e056bbb2ea92e6cdd40def).

This bug had been lurking in the connection pool code for a very long time. Although it wasn't nearly as obvious before setBindRequest, the connection pool was actually double-authenticating each new connection (once in the server set and again using the bind request). Even without setBindRequest, this would have had at least a small performance impact from the unnecessary extra bind request.
